edx/blockstore
# Switch to Github Actions and remove constraints on the `cryptography` and `mysqlclient` packages.

## Description

We have a [constraint](https://github.com/edx/blockstore/blob/0424dc6/requirements/constraints.txt#L29-L36) on the `cryptography` package because the CircleCI tests use `alpine` docker image on which a Rust dependency is unavailable.

To be able to install [blockstore as a package](https://github.com/edx/blockstore/pull/97) in edx-platform, this constraint needs to be removed (edx-platform is on the latest version of the `cryptography` package).

CircleCI provides [custom docker images](https://circleci.com/docs/2.0/circleci-images/) but to be able to use Docker Hub images, to get around the rate-limiting, tokens will need to be set up (I am not clear why we haven't hit that for the `alpine` image). Instead it is simpler to switch to Github actions which is already being used for two workflows in this repo and which a number of other repos in the edx org have switched to.

## Ticket
* [BOM-2643](https://openedx.atlassian.net/browse/BOM-2643)
* [FAL-2231](https://tasks.opencraft.com/browse/FAL-2231)

## Author Comments, Concerns, and Open Questions

## Test Instructions

* Verify the Github Actions job run the relevant tests and quality checks.
* Checkout the branch locally.
* Run
```
make destroy
make build
make dev.up
make dev.provision
make blockstore-shell

# In the shell
make test
```

## TODOs
If anything isn't yet done, list it here
- [ ] Get Github Actions CI job green.
- [ ] Remove constraints on the `cryptography` and `mysqlclient` packages.
